### ISSUE TITLE
64 bit builds do not support joysticks

### ISSUE THREAD

**Issue by almeath** [alunbestor/Boxer#102]
I have tested a range of joysticks and controllers in 64 bit builds of Boxer, both the 64 bit master and Madd the Sane's custom branch.

I cannot get a single joystick to respond properly. While the 'fire' buttons will always work, the actual control stick will never respond to x/y axis input. However, when I use the same joysticks in the standard 32 bit version 1.4 build they all work fine. 

Also, standard d-button game controllers continue to work in both 64 bit and 32 bit, it is just joysticks that are affected in 64 bit.

**Comment by augmentedfourth** [comment]
> Also, standard d-button game controllers continue to work in both 64 bit and 32 bit, it is just joysticks that are affected in 64 bit.

I thought the Gravis GamePad Pro I'm trying to use was a "d-button game controller." It only has 8 directional positions plus the center at rest; there's no gradual movement between directions like the upright joysticks of old.
https://images-na.ssl-images-amazon.com/images/I/41YX7YR374L._AC_.jpg

My gamepad has problems in the same way as described, though... buttons all work, but no directional controlling has any effect on the game.

Do you have a suggestion for a gamepad that would work with a 64-bit build? This is honestly the only thing keeping me from upgrading to Catalina/10.15.

**Comment by almeath** [comment]
I have been able to get a Logitech Dual Action pad to work:

https://www.logitech.com/lang/pdf/dual_action.pdf

However I tried an old Macally iShock and that failed. So it appears that it is entirely hit-and-miss as to what will be compatible. I would suggest you pick up a cheap Logitech controller on eBay and see how you go.

The other option, which I tested and confirmed to be working, is to use a controller reprogramming app (i.e. Controller Mate, USB Overdrive, Controllers, GamePad Companion etc.) to re-map your gamepad buttons to the DOS keyboard equivalents in your games. The only annoyance there is having to keep re-mapping keys for different games, unless you use game application wrappers that are recognized as individual apps by the reprogramming app. It gets a bit fiddly if you are not into that kind of setup.

**Comment by augmentedfourth** [comment]
Ah, yes. I've used Enjoyable before. I'll give that a shot. I'd much rather keep with the Gravis Gamepad style/shape that's been comfortable for me ever since I played One Must Fall (the blocky weird beta version, not the cool robots in OMF:2097) on an actual MS-DOS machine.

**Comment by almeath** [comment]
Fair enough - good luck there. I managed to get an old Microsoft Sidewinder working in the same way, by using Controller Mate to re-map the keys for games like Red Baron and Aces Over the Pacific. Controller Mate is not yet Catalina compatible, but there are lots of options out there for macOS in general.

**Comment by augmentedfourth** [comment]
Enjoyable does make the joystick control work better (because, of course, it's merely emulating the keyboard). But now that I can actually play, I find the jerkiness of the game really hard to play through.

**Comment by almeath** [comment]
Controller Mate IV has a steep learning curve but would give you total control over all elements of your input device, including the sensitivity of key presses and pad/joystick movements.

One last option would be to switch to DOSBox for games that need precise joystick control. I have a pre-built app wrapper you can try, if you email me on my Github username at the domain eftel.com

**Comment by augmentedfourth** [comment]
Yeah, it's not about joystick control at this point. I hadn't really used Boxer 64-bit yet because I hadn't gotten my gamepad to work, but now that it's working there are other problems. Probably not related to this thread, but I'll create a new issue.

**Comment by freecodecampster** [comment]
Hi almeath, I'm using a Mayflash fighting stick attached to a Wiimote which works with both OpenEmu and RetroArch running DosBox core on macOS Catalina.

A working example with Speedball II and RetroArch. Install the DosBox core, load the content - first setup.exe. Choose joystick as the control method. Then run sb2 at the command prompt. The dos game works perfectly with the controller.

Trying Boxer on macOS Catalina. The inspector shows a joystick is connected but button presses do nothing. Controllers just don't seem to work.

Boxer in theory should be able to be made to work with controllers on Catalina. It's a shame as it has a great UI. I'm not a great fan of RetroArch's long menus. But at least it's a working, though slightly more tedious option.

**Comment by almeath** [comment]
Yes, I have had no issues with getting joysticks and game pads to work in DOSBox and Retroarch (separately). I have never tried to use the DOSBox core in Retroarch, but it sounds like another viable option.

The thing to remember with Boxer is that official development of it effectively ceased over 5 years ago, so most of its code is frozen in time at that point. There has been plenty of unofficial patching going on since then, but nothing so far has fixed the known issue with the joystick x/y axis. Some game pads remain incompatible as well. If game pads and joysticks are essential for your DOS gaming then you are best off learning how to use vanilla DOSBox or SVN.

**Comment by augmentedfourth** [comment]
I've been playing with macOS builds (including local builds from source) of vanilla DOSBox and DOSBox-X, and finding even less success than 64-bit alphas of Boxer. @almeath, how did you end up with "no issues" in that scenario? Did you use a Homebrew release, a pre-built binary, or something else?

**Comment by almeath** [comment]
Perhaps you can try the DOSBox SVN snapshot maintained by Dominus from the VOGONS forum:

https://www.dropbox.com/s/kbrf5elbcu66kst/Dosbox-Snapshot.dmg

It’s a 'fat binary' supporting x64, i386 and PPC. 

I have tried compiling DOSBox in macOS Mojave and always have problems getting all the required frameworks to load properly. I think Dominus still compiles in High Sierra.

**Comment by aminta** [comment]
> Perhaps you can try the DOSBox SVN snapshot maintained by Dominus from the VOGONS forum:
> 
> https://www.dropbox.com/s/kbrf5elbcu66kst/Dosbox-Snapshot.dmg
> 
> It’s a 'fat binary' supporting x64, i386 and PPC.
> 
> I have tried compiling DOSBox in macOS Mojave and always have problems getting all the required frameworks to load properly. I think Dominus still compiles in High Sierra.

Thanks!

**Comment by chrismaaz** [comment]
Xbox One Bluetooth Controller works for me, but it seems to depend on what game you are playing.
